Feature/manage instance service

ZeroPointEnergy opened the pull request:
This is an optional setting to manage the db2 instances service with systemd. It is disabled by default to retain backward compatibility to previous configurations.

crayfishx commented on manifests/instance.pp:
```diff
@@ -46,6 +47,35 @@
+      enable    => true,
+      subscribe => [
+        Exec['db2_systemd_daemon_reload'],
+        Db2_instance[$instance_user],
```
This suggests that the service needs to be restarted after adding an instance, dcs, catalog or database.... Is this really the case?

ZeroPointEnergy replied:
I thought so, but now that you ask I will check this with the database specialists so we get the behaviour right.

crayfishx commented on manifests/instance.pp:
```diff
@@ -46,6 +47,35 @@
+      enable    => true,
+      subscribe => [
+        Exec['db2_systemd_daemon_reload'],
+        Db2_instance[$instance_user],
```
@ZeroPointEnergy any update on this?  Id like to merge your change but we need to clarify and update the pr if necessary.   Thanks

ZeroPointEnergy commented on manifests/instance.pp:
```diff
@@ -46,6 +47,35 @@
+      enable    => true,
+      subscribe => [
+        Exec['db2_systemd_daemon_reload'],
+        Db2_instance[$instance_user],
```
@crayfishx sorry for the delay, we where able to get some more information on this. The DB2 specialists recommend that the instance does not get started when new nodes, databases and dcs are catalogued, but that the instance has to be running already.

Also another question came up about if the service needs to be managed at all or if only the service file is required. I will close the current PR if that is ok and create a new one once this is sorted.

ZeroPointEnergy commented:
Ok, never mind, I just fixed it in the current PR. I also added the possibility to set the service ensure and enable parameters per instance. The service is enabled by default to make sure it starts up on server reboot, but puppet does not try to start it by default (so it does not interfere with maintenance etc).

Please let me know if there are additional concerns.

ZeroPointEnergy commented:
@crayfishx is it possible that we can get this PR merged and a new release? Let me know if there is anything missing.
